torch/nn/functional.py

```diff
@@ -2815,6 +2815,26 @@ def multi_margin_loss(input, target, p=1, margin=1., weight=None, size_average=N
     torch.Size([1, 1, 12, 12])
 """)
 
+pixel_unshuffle = _add_docstr(torch.pixel_unshuffle, r"""
+pixel_unshuffle(input, downscale_factor) -> Tensor
+
+Rearranges elements in a tensor of shape :math:`(*, C, H \times r, W \times r)` to a
```

Comment:
do we use `*` in the docs consistently how you've used it here (i.e. correctly?).  I'm particularly asking about the case where `*` translates to no dimensions.

Reply:
Hmm good question! From the [nn.Linear](https://pytorch.org/docs/stable/generated/torch.nn.Linear.html) docs:

"Input: `(N, *, H_{in})`, where `*` means any number of additional dimensions and `H_{in}` = in_features"

This description indicates to me that `N` and `H_{in}` are required dimensions, and `*` can be any number of additional dimensions in between those two. However, surprisingly to me, `nn.Linear` does accept an input shape of `(H_{in},)`. It seems that the `*` functions like "0 or more of the previous" (i.e. the Kleene star). I think it is the use of the word "additional" that makes this surprising to me, but maybe this is only a problem for me.

Similarly in [nn.L1Loss](https://pytorch.org/docs/stable/generated/torch.nn.L1Loss.html):

"Input: `(N, *)` where `*` means, any number of additional dimensions"

`nn.L1Loss` does indeed accept tensors of 0 dimensions, so again the `*` functions like a Kleene star.

**TL;DR: To match what's been done in the two referenced modules, the line here should change to: `(N, *, C, H \times r, W \times r)`. Similar updates should be done in `PixelShuffle` for consistency.**

Reply:
We do typically use `*` for 0 or more dimensions. See, for example, the torch.svd documentation: https://pytorch.org/docs/master/generated/torch.svd.html?highlight=svd#torch.svd. We're probably inconsistent about this in our documentation. 

I could see `N*` or `*` as appropriate. I wouldn't want to use `(N, *, C...)` unless zero or more dimensions could be inserted between a required dimension `N` and a required dimension `C`.

Reply:
> I could see N* or * as appropriate. I wouldn't want to use (N, *, C...) unless zero or more dimensions could be inserted between a required dimension N and a required dimension C.

I agree with this as it matches my intuition, and I think we should make a note to eventually make this consistent across torch.nn modules (including `nn.Linear` and `nn.L1Loss`).

Reply:
As far as this PR is concerned, I will add a note to the docs indicating what * represents here, with similar phrasing to that of `torch.svd`.